services/dynamic-sidecar/tests/unit/test_api_containers.py

```diff
@@ -450,3 +450,21 @@ async def test_containers_entrypoint_name_containers_not_started(
     assert response.json() == {
         "detail": "No container found for network=entrypoint_container_network"
     }
+
+
+async def test_containers_restart_after_starting(
```

Comment:
this function does not really tell me what it does. 
```suggestion
async def test_containers_restart(
```
Also you do not check whether the container has effectively restarted right? you only check that the fastapi returns without error here.

Reply:
very good point, will add this check